lib/facter/self_service.rb

```diff
@@ -108,20 +108,18 @@
     { S0021: Facter.value(:memory)['system']['capacity'].to_f <= 90 }
   end
 
-  chunk(:S0036) do
-    next unless PuppetSelfService.replica? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler? || PuppetSelfService.primary?
-    str = IO.read('/etc/puppetlabs/puppetserver/conf.d/pe-puppet-server.conf')
-    max_queued_requests = str.match(%r{max-queued-requests: (\d+)})
-    if max_queued_requests.nil?
-      { S0036: true }
-    else
-      { S0036: max_queued_requests[1].to_i < 150 }
-    end
-  end
   chunk(:S0030) do
     # check for use_cached_catalog logic flip as false is the desired state
     { S0030: !Puppet.settings['use_cached_catalog'] }
   end
+
+  chunk(:S0032) do
+    # MCollective/ActiveMQ enabled/exists?
+    mco_check = PuppetSelfService.service_running_enabled('mcollective') || PuppetSelfService.service_file_exist?('mcollective') || PuppetSelfService.get_resource('cron', 'pe-mcollective-metadata')
+    # amq_check = PuppetSelfService.service_running_enabled('pe-activemq') || PuppetSelfService.service_file_exist?('pe-activemq')
+    { S0032: mco_check.nil? }
```

Comment:
Shouldn't this be checking for a boolean instead of `nil?`? The one below moves the `nil?` logic to the `mco_check` and uses a boolean check in the output.

```
mco_check = PuppetSelfService.service_running_enabled('mcollective') || PuppetSelfService.service_file_exist?('mcollective') || PuppetSelfService.get_resource('cron', 'pe-mcollective-metadata').nil?
{ S0032: !mco_check }
```

Reply:
Good catch! @jarretlavallee @MartyEwings should the amq_check be moved to its own chunk or should I just move that amq stuff up into the mco_check and report false if mco or amq is still lingering around?